calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,149 @@
-
 from netaddr import IPAddress, IPNetwork, AddrFormatError
+
+from logutils import configure_logger
 from pycalico import netns
 from pycalico.ipam import IPAMClient, SequentialAssignment
 from pycalico.netns import Namespace
-from pycalico.datastore_datatypes import Rules
+from pycalico.datastore_datatypes import Rules, IPPool
 from pycalico.datastore import IF_PREFIX, DatastoreClient
 from pycalico.datastore_errors import PoolNotFound
 
-print_stderr = functools.partial(print, file=sys.stderr)
+logger = logging.getLogger(__name__)
 
-
-input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
-input_json = json.loads(input_)
-
-# Append to existing env, to avoid losing PATH etc.
-# TODO-PAT: This shouldn't be hardcoded
-env = os.environ.copy()
 ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
-env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
 
-# PROFILE_LABEL = 'CALICO_PROFILE'
-# ETCD_PROFILE_PATH = '/calico/'
+env, INPUT_JSON = {}, {}
 
 ORCHESTRATOR_ID = "rkt"
 HOSTNAME = socket.gethostname()
-NETNS_ROOT= '/var/lib/rkt/pods/run'
+NETNS_ROOT = '/var/lib/rkt/pods/run'
 
 def main():
     mode = env['CNI_COMMAND']
 
     if mode == 'init':
-        print_stderr('No initialization work to perform')
+        logger.info('No initialization work to perform')
     elif mode == 'ADD':
-        print_stderr('Executing Calico pod-creation plugin')
-        create(
-            container_id=env['CNI_CONTAINERID']
-            )
+        logger.info('Executing Calico pod-creation plugin')
+        create(container_id=env['CNI_CONTAINERID'])
     elif mode == 'DEL':
-        print_stderr('Executing Calico pod-deletion plugin')
-        delete(
-            container_id=env['CNI_CONTAINERID']
-            )
+        logger.info('Executing Calico pod-deletion plugin')
+        delete(container_id=env['CNI_CONTAINERID'])
 
 def create(container_id):
     """"Handle rkt pod-create event."""
-    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    logger.info('Configuring pod %s' % container_id)
     netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
     _datastore_client = IPAMClient()
 
     try:
         endpoint, ip = _create_calico_endpoint(container_id=container_id,
-                                            netns_path=netns_path,
-                                            client=_datastore_client)
+                                               netns_path=netns_path,
+                                               client=_datastore_client)
 
-        _create_profile(endpoint=endpoint, 
-                        profile_name=input_json['name'], 
+        _create_profile(endpoint=endpoint,
+                        profile_name=INPUT_JSON['name'],
                         ip=ip,
                         client=_datastore_client)
     except CalledProcessError as e:
-        print_stderr('Error code %d creating pod networking: %s\n%s' % (
+        logger.info('Error code %d creating pod networking: %s\n%s' % (
```

Comment:
logs in exception blocks should use logger.exception(..) rather than logger.info(..).

In general, make sure to be thinking about logging levels for individual messages.  Some are info level, but some are only useful for debug level and might become repetitive / cluttered if at info level, etc.